resources/type-system/inference.md

```diff
@@ -774,7 +777,10 @@ clauses need be tried.
 - If `P` is a legacy type `P0*` then the match holds under constraint set `C`:
   - Only if `P0` is a subtype match for `Q` under constraint set `C`.
 - If `Q` is a legacy type `Q0*` then the match holds under constraint set `C`:
-  - Only if `P` is a subtype match for `Q?` under constraint set `C`.
+  - If `P` is `dynamic` or `void` and `P` is a subtype match for `Q` under
+    constraint set `C`.
```

Comment:
It seems that this rule makes the algorithm recurse infinitely.

Reply:
Sorry, there was a typo in the original rule that I carried over to the new rule.  Fixed in both rules now.